lighthouse-core/audits/metrics.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('./audit');
+
+class Metrics extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      name: 'metrics',
+      informative: true,
+      description: 'Metrics',
+      helpText: 'Collects all available metrics.',
+      requiredArtifacts: ['traces', 'devtoolsLogs'],
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static async audit(artifacts, context) {
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
+    const devtoolsLog = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
+    const metricComputationData = {trace, devtoolsLog, settings: context.settings};
+
+    const traceOfTab = await artifacts.requestTraceOfTab(trace);
+    const firstContentfulPaint = await artifacts.requestFirstContentfulPaint(metricComputationData);
+    const firstMeaningfulPaint = await artifacts.requestFirstMeaningfulPaint(metricComputationData);
+    const timeToInteractive = await artifacts.requestConsistentlyInteractive(metricComputationData);
+    const metricsMap = {firstContentfulPaint, firstMeaningfulPaint, timeToInteractive};
```

Comment:
move this `metricsMap ` line to right above the `for` loop. move the `const metrics` line up higher.

Reply:
done